src/Microsoft.AspNetCore.Mvc.Razor/Compilation/CompiledViewDescriptor.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
+{
+    public struct CompiledViewDescriptor
+    {
+        public string RelativePath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="RazorViewAttribute"/> decorating the sview.
+        /// </summary>
+        public RazorViewAttribute ViewAttribute { get; set; }
```

Comment:
Ah I see what you're doing. Consider just making this class abstract `CompiledItemDescriptor` and having two concrete subclasses (view + page)